### ISSUE TITLE
Make it real easy to open an issue with crash reports and logs

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#1657]
Imagine a `flutter report-bug` command. It could grab any crash report from iOS, any recent logs, the flutter version, the Dart version, and then send it off to GitHub as an issue.

We want to make it as simple as

Example UI:

```
$ flutter report-bug
Sorry you ran into trouble, but we appreciate the bug report.

This command collects potentially relevant information and opens a public bug report for the Flutter team as a GitHub issue. Please review the information before you confirm.

Flutter version: ...
Dart version: ...
OS: mac
Crash file: path/to/file
Log: path/to/file or print out what we will send

OK to make this information public as a GitHub issue? [y/N]
```

**Comment by Hixie** [comment]
cc @tvolkert who might be interested in this

**Comment by tvolkert** [comment]
I was actually thinking about this a few days ago.  I think it works best as a flag to `flutter run`, since that lets you capture potentially important information in the building and deployment of the app.

**Comment by tvolkert** [comment]
`flutter run --bugreport`

**Comment by zanderso** [comment]
@jmagman's work [here](https://github.com/flutter/flutter/commit/7d8f82051b53baf64e96066434a5bf98305d8947) in this area is what we decided to do for this.

**Comment by jmagman** [comment]
Let's reopen this.  https://github.com/flutter/flutter/pull/45360 is triggered for `flutter` crashes, but it would be nice to have something the user could trigger if they are hitting a non-crashing bug.  We should be able to reuse a lot of that code.

**Comment by bkonyi** [comment]
> Let's reopen this. #45360 is triggered for `flutter` crashes, but it would be nice to have something the user could trigger if they are hitting a non-crashing bug. We should be able to reuse a lot of that code.

What sort of non-crashing bugs were you thinking of? Just general failures in the tool (e.g., no device found)?

**Comment by jmagman** [comment]
Yes, or even bugs in Flutter itself "I launch my app and the screen is black" that would collect verbose logs, doctor, etc

**Comment by bkonyi** [comment]
I don't think this is something that the tool team will have time to work on, and I can't think of a good way to implement this in the UI. Going to close this as not planned for now.
